cmd/ztest/ztest.c

```diff
@@ -6744,6 +6759,34 @@ ztest_dataset_close(int d)
 	ztest_zd_fini(zd);
 }
 
+/* ARGSUSED */
+static int
+ztest_replay_zil_cb(const char *name, void *arg)
+{
+	objset_t *os;
+	ztest_ds_t *zdtmp;
+
+	VERIFY0(ztest_dmu_objset_own(name, DMU_OST_ANY, B_TRUE,
+	    B_TRUE, FTAG, &os));
+
+	if (dmu_objset_zil(os)->zl_parse_lr_count != 0 &&
+	    ztest_opts.zo_verbose >= 4) {
+		(void) printf("replaying logs for '%s' (%llu records)\n", name,
+		    (unsigned long long)dmu_objset_zil(os)->zl_parse_lr_count);
+	}
```

Comment:
Can we move this logging after `zil_replay()` and log it in the same format used by `ztest_dataset_open()`.

```c
        if (ztest_opts.zo_verbose >= 6)
                (void) printf("%s replay %llu blocks, %llu records, seq %llu\n",
                    zd->zd_name,
                    (u_longlong_t)zilog->zl_parse_blk_count,
                    (u_longlong_t)zilog->zl_parse_lr_count,
                    (u_longlong_t)zilog->zl_replaying_seq);
```